### DISCUSSION TITLE
How to Run from the project

### DISCUSSION THREAD

**Discussion by Alik2015** [simpleidserver/SimpleIdServer#586] [General]
I have cloned the repo and wish to know the order of steps to run the id server from the sln.  I tried to run idserver/startup and am faced with a certificate window and when selecting any am getting 404.

I wish to have the server pop up the login page etc so that I can see how it works and which views to change to customize.  Also wish to try it with MySQL which I am guessing should be a case of creating and running migrations manually against MySQL Server. 

There does not seem to be much documentation about how to run from the project.

Appreciate any tips.

**Comment by simpleidserver** [comment]
Hello,

By default, SimpleIdServer is configured to accept client certificates because the client authentication method 'tls_client_auth' is enabled by default and requires the client certificate. You can click on the 'Cancel' button to proceed with the authentication :).

If you wish to support MySQL, please follow the steps below:

1. Install the NuGet package `MySql.EntityFrameworkCore`.
2. Edit the `StorageTypes` class and add the value 'MYSQL'.
3. Edit the 'Program.cs' file, and update the 'ConfigureStorage' method as follows:

```        
case StorageTypes.POSTGRE:
            var migrationsAssembly = typeof(Program).GetTypeInfo().Assembly.GetName().Name;
            b.UseMySQL(conf.ConnectionString, o =>
            {
                o.MigrationsAssembly(migrationsAssembly);
                o.UseQuerySplittingBehavior(QuerySplittingBehavior.SplitQuery);
            });
            break;
```
4. Execute this command line to create the migration scripts: `dotnet ef migrations add Init`.
5. Edit the `appsettings.json` file, update the property `StorageConfiguration.Type` to `MYSQL`, and update the ConnectionString.

If you have any further questions or need assistance, please feel free to ask.

**Reply by Alik2015** [reply]
Very much appreciate the answer and will try.

**Reply by Alik2015** [reply]
Using the above method also tried to update the SCIM project and but it cannot create schemas in MYSQL so tried the following in SCIM Startup program.cs
```

  case StorageTypes.MYSQL:
                        var migrationsAssembly = typeof(Program).GetTypeInfo().Assembly.GetName().Name;
                        options.UseMySql(conf.ConnectionString, ServerVersion.AutoDetect(conf.ConnectionString), o =>
                        {
                        o.MigrationsAssembly(migrationsAssembly);
                        o.UseQuerySplittingBehavior(QuerySplittingBehavior.SplitQuery);
                        o.SchemaBehavior(Pomelo.EntityFrameworkCore.MySql.Infrastructure.MySqlSchemaBehavior.Ignore);                        });
                        break;
`
```

I which seemed to create the db in MYSQL Along with the tables and I could register clients.  However, when using the interface I see an error:

![screen](https://github.com/simpleidserver/SimpleIdServer/assets/31630746/eccfb2ec-6f39-4d80-9d64-e1c81ba3358d)

**Reply by simpleidserver** [reply]
Unfortunately, we are experiencing the same issue in the demo application. The administration website is attempting to load an unknown CSS file. However, the website should still function. Can you please try navigating to the different tabs `Clients` and `Scopes`?

**Reply by Alik2015** [reply]
For anyone having the ui issue I can confirm UI seems to be working fine.

**Reply by Alik2015** [reply]
Hi I added MySQL Support and pull request was merged a few weeks back.  Just wondering if you have removed the MySQL modifications in the StorageTypes and the migrations as when pulling the new updates I don't see MySQL as storage type anymore?

**Reply by simpleidserver** [reply]
The MYSQL support is still present in the `master` branch :)

Enumeration : 
https://github.com/simpleidserver/SimpleIdServer/blob/28bfecf5bb8370e53a4faaa265de077519d2fb08/src/Scim/SimpleIdServer.Scim.Startup/Configurations/StorageConfiguration.cs#L16

Identity Server Storage : 
https://github.com/simpleidserver/SimpleIdServer/blob/28bfecf5bb8370e53a4faaa265de077519d2fb08/src/IdServer/SimpleIdServer.IdServer.Startup/Program.cs#L258

Identity Server configuration : 
https://github.com/simpleidserver/SimpleIdServer/blob/28bfecf5bb8370e53a4faaa265de077519d2fb08/src/IdServer/SimpleIdServer.IdServer.Startup/Program.cs#L228

**Comment by Alik2015** [comment]
What I am trying to figure is what projects I need to run in order to just have the id server part working.

Basically currently using OpenIddict and wish to replace with SimpleIdServer.

So questions are as follows:

When a client app is set up.  Where is the login UI that I can modify example login page?

I am running only the src\idserver\impleIdServer.IdServer.Startup project.  Is this enough to get started, or do I need to run other projects alongside it?

I understand at this time there is no tutorials online or clear documentation which could be due to limited time etc, so it is easier just to ask and any guidence is appeciated.

**Comment by simpleidserver** [comment]
Hello,

Currently, there is no documentation available that explains how to use the various NuGet Packages. However, you can utilize our DOTNET TEMPLATE project to create a local Identity Server project. You can refer to the documentation at https://simpleidserver.com/docs/installation#dotnet-template for detailed instructions.

This project includes the necessary views and configurations. Please note that the documentation will be enhanced in future releases to provide guidance on using the different NuGet packages.

Best regards,

SID

**Comment by simpleidserver** [comment]
The documentation will be updated to explain how to use the Nuget package, Ticket is #591 : https://github.com/simpleidserver/SimpleIdServer/issues/591
